[GabrielCWT] iP

GabrielCWT opened the pull request:
# Joe Chatbot
* This is **the** chatbot named Joe
> "Your mind is for having ideas, not holding them.” – David Allen
He can do the following:
1. Add todos
2. Delete todos
3. Store it in a file
4. Mark/Unmark todos
5. Look up ~~tasks~~ todos

If you are _interested_, [click here](https://github.com/GabrielCWT/ip)

Everything has been abstracted away! Take a look at the `main` method
```java
    public static void main(String[] args) {
        new Joe().run();
    }
```

## TODOs Week 4 🤡 
[x] GFMD in PR
[ ] Review PR
[ ] Add increments as branches

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

GabrielCWT committed:
chore: Initial refactor

GabrielCWT committed:
feat: Implement initial skeleton

GabrielCWT committed:
refactor: Abstract greeting and farewell

GabrielCWT committed:
feat: Implement echoing of user inputs

GabrielCWT committed:
feat: Implement list function

GabrielCWT committed:
feat: Implement task class

GabrielCWT committed:
refactor: Refactor mark/unmark to toggle

GabrielCWT committed:
feat: Implement todo list

GabrielCWT committed:
chore: Fix indentation

GabrielCWT committed:
feat: Add different types of tasks

GabrielCWT committed:
refactor: Rename task to tasktodo

GabrielCWT committed:
fix: Format issues

GabrielCWT committed:
feat: Implement tests

GabrielCWT committed:
feat: Include unmark test

GabrielCWT committed:
feat: Implement handling of invalid inputs

GabrielCWT committed:
feat: Implement delete function

GabrielCWT committed:
refactor: Use arraylist instead of array for store

GabrielCWT committed:
refactor: Rename constants to all caps naming convention

GabrielCWT committed:
feat: Implement save string method in all tasks

GabrielCWT committed:
feat: Implement save/load tasks from files

GabrielCWT committed:
Merge branch 'branch-Level-7'

GabrielCWT committed:
refactor: Store dates using Date object instead of string

GabrielCWT committed:
feat: Implement a help command

GabrielCWT committed:
Merge branch 'branch-Level-8'

GabrielCWT committed:
refactor: Separate app into different classes

GabrielCWT committed:
Merge branch 'A-MoreOOP'

GabrielCWT committed:
chore: Move files into package

GabrielCWT committed:
Merge branch 'add-gradle-support'

GabrielCWT committed:
fix: Replace use of getLast in store

GabrielCWT committed:
refactor: Move files into their own packages

GabrielCWT committed:
refactor: Abstract all parsing logic into parser

GabrielCWT committed:
refactor: Move scanner to joe class

GabrielCWT committed:
fix: Fix accessing out of bounds array

GabrielCWT committed:
feat: Implement controller stub

GabrielCWT committed:
feat: Implement ParserTest

GabrielCWT committed:
feat: Implement ControllerTest

GabrielCWT committed:
chore: Update gradle config

GabrielCWT committed:
docs: Add javadoc for functions

GabrielCWT committed:
Merge branch 'branch-A-JavaDoc'

GabrielCWT committed:
chore: Format files

GabrielCWT committed:
Merge branch 'branch-A-CodingStandard'

GabrielCWT committed:
feat: Implement find feature

GabrielCWT committed:
chore: Make utility objects final

GabrielCWT committed:
Merge branch 'branch-Level-9'

GabrielCWT committed:
docs: Include header description for classes

FooChao reviewed (commented):
Amazing job ensuring your method and variable names follow the naming convention. I couldn't find any errors with regards to code quality.

tata32000 commented on src/main/java/joe/parser/Parser.java:
```diff
@@ -0,0 +1,74 @@
+import joe.controller.Controller;
+import joe.ui.Ui;
+
+public class Parser<C extends Controller> {
```
Maybe can rename C to other meaningful name?

tata32000 commented on src/main/java/joe/task/Task.java:
```diff
@@ -0,0 +1,53 @@
+    /**
+     * Prints the task with its status.
+     */
+    public void printTask() {
```
Should this be renamed printTaskStatus?

tata32000 commented on src/main/java/joe/ui/Ui.java:
```diff
@@ -0,0 +1,112 @@
+    }
+
+    public void printHelp() {
+        System.out.println("Here are the commands you can use:");
```
Should this be formatted before print so you won't have so many System.out.println?

GabrielCWT committed:
feat: Add files for GUI feature

GabrielCWT committed:
refactor: Refactor files to use GUI

GabrielCWT committed:
refactor: Refactor output messages to use GUI

GabrielCWT committed:
chore: Update build file

hpcman commented on src/main/java/joe/Joe.java:
```diff
@@ -0,0 +1,41 @@
+
+    /**
+     * Generates a response for the user's chat message.
+     */
```
Missing @return for JavaDoc. I think you missed this for some JavaDoc

hpcman commented on src/main/java/joe/controller/Controller.java:
```diff
@@ -0,0 +1,178 @@
+     * 
+     * @param index
+     *            The index of the task to be marked as done.
+     */
```
I think you shouldn't put @param documentation in the next line. It is better to put it in same line unless too long. This suggestion applies to another JavaDoc

hpcman reviewed (commented):
Overall LGTM. Just some nit in some file related to JavaDoc

GabrielCWT committed:
chore: Remove unused imports and code

GabrielCWT committed:
feat: Implement assertions for Controller and Parser

The functions assume that the inputs passed in are not null.

There could be some unhandled behaviour due to these assumptions.

Assertions have been implemented to ensure that other functions do not
break due to null inputs.

GabrielCWT committed:
chore: Formatting

GabrielCWT committed:
refactor: Abstract parser code to helper functions

GabrielCWT committed:
refactor: Refactor parser to use constant variables

GabrielCWT committed:
refactor: Refactor task completion string to be a constant

GabrielCWT committed:
feat: Add checks for invalid index

GabrielCWT committed:
refactor: Abstract long functions into helper functions in Storage

GabrielCWT committed:
refactor: Refactor task type strings to be constants

GabrielCWT committed:
chore: Change Task to be abstract

GabrielCWT committed:
docs: Add javadoc to public functions

GabrielCWT committed:
chore: Update comment formatting

GabrielCWT committed:
Merge pull request #1 from GabrielCWT/branch-A-Assertions

Implement Assertions

GabrielCWT committed:
Merge branch 'master' into branch-A-CodeQuality

GabrielCWT committed:
Merge pull request #2 from GabrielCWT/branch-A-CodeQuality

Improve code quality

GabrielCWT committed:
feat: Implement postpone feature

GabrielCWT committed:
fix: Reduce size of user icon

GabrielCWT committed:
feat: Implement auto-saving on close

GabrielCWT committed:
docs: Update README

GabrielCWT committed:
fix: Fix bye command not exiting app

GabrielCWT committed:
refactor: Remove unnecessary generic for parser

GabrielCWT committed:
fix: Implement UiStub for ControllerTest

GabrielCWT committed:
docs: Update README

GabrielCWT committed:
chore: Add JavaDoc for public methods

GabrielCWT committed:
fix: Fix mark/unmarking of tasks

GabrielCWT committed:
chore: Make date error message more specific

GabrielCWT committed:
fix: Fix printed list for find message

GabrielCWT committed:
docs: Update README
